sessionctx/variable/sysvar.go

```diff
@@ -1501,6 +1501,12 @@ var defaultSysVars = []*SysVar{
 	{Scope: ScopeGlobal, Name: TiDBGCLifetime, Value: "10m0s", Type: TypeDuration, MinValue: int64(time.Minute * 10), MaxValue: math.MaxInt64},
 	{Scope: ScopeGlobal, Name: TiDBGCConcurrency, Value: "-1", Type: TypeInt, MinValue: 1, MaxValue: 128, AllowAutoValue: true},
 	{Scope: ScopeGlobal, Name: TiDBGCScanLockMode, Value: "PHYSICAL", Type: TypeEnum, PossibleValues: []string{"PHYSICAL", "LEGACY"}},
+
+	// See https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_temptable_max_ram
+	{Scope: ScopeGlobal, Name: TempTableMaxRAM, Value: strconv.Itoa(DefTempTableMaxRAM), Type: TypeInt, MinValue: 2097152, MaxValue: math.MaxInt64, SetGlobal: func(s *SessionVars, val string) error {
```

Comment:
For MySQL, the max value is MaxUint64.

Reply:
It's okay to not follow strictly, but there is an issue in that the internal value is a signed int32, but the sysvar allows up to signed int64. The two types should align.

Signed int32 only allows up to 2GB temporary tables, which is "okay", but there might be some cases where it needs to be larger.

Reply:
It is still set to have a maximum value of `math.MaxInt64`, but the internal value is an int32.

Reply:
I think it's better to be conservative.
We can update the value to be larger anytime as long as we'd like to, but if we set it's initial value huge and turn down the value in a future release, there will be a compatibility issue.
And now I take @morgo 's advise and set it to `math.MaxInt32`